### ISSUE TITLE
dynamic keyword failure

### ISSUE THREAD

**Issue by fredericDelaporte** [maca88/AsyncGenerator#113]
Trying to generate async counterparts for method using the C# `dynamic` keyword seems to cause the async generator to get somewhat lost, no more recognizing mundane methods like `That` on `Assert`.

I have not found a way to reproduce this with the latest version, but I am not able to check the latest version would work either due to #112.

The trouble is seen with AsyncGenerator 0.8.2.7 run on [this branch](https://github.com/fredericDelaporte/nhibernate-core/tree/dynamic-component).

It fails with an exception: `System.InvalidOperationException: Unable to find symbol for node That inside function NHibernate.Test.EntityModeTest.Map.Basic.DynamicClassFixture.ShouldWorkWithLinqAndDynamics()`, from `ProjectAnalyzer.<ScanAllMethodReferenceLocations>d__78.MoveNext()`.

Log:
```
16:03:16,485 INFO  AsyncCodeGenerator:0 - Opening project 'E:\Projets\nhibernate\nhibernate-core\src\NHibernate.Test\NHibernate.Test.csproj' started
16:03:18,563 INFO  AsyncCodeGenerator:0 - Opening project 'E:\Projets\nhibernate\nhibernate-core\src\NHibernate.Test\NHibernate.Test.csproj' completed
16:03:18,563 INFO  AsyncCodeGenerator:0 - Configuring project prior analyzation started
16:03:18,578 INFO  AsyncCodeGenerator:0 - Configuring project prior analyzation completed
16:03:18,578 INFO  AsyncCodeGenerator:0 - Compiling project 'NHibernate.Test' started
16:03:21,890 INFO  AsyncCodeGenerator:0 - Compiling project 'NHibernate.Test' completed
16:03:21,890 INFO  AsyncCodeGenerator:0 - Initializing registered plugins for project 'NHibernate.Test' started
16:03:21,906 INFO  AsyncCodeGenerator:0 - Initializing registered plugins for project 'NHibernate.Test' completed
16:03:21,906 INFO  AsyncCodeGenerator:0 - Analyzing project 'NHibernate.Test' started
16:03:21,921 INFO  ProjectAnalyzer:0 - Parsing documents started
16:03:21,968 INFO  ProjectAnalyzer:0 - Parsing documents completed
16:03:21,968 INFO  ProjectAnalyzer:0 - Pre-analyzing documents started
16:03:23,296 INFO  ProjectAnalyzer:0 - Pre-analyzing documents completed
16:03:23,296 INFO  ProjectAnalyzer:0 - Scanning references started
16:03:50,055 FATAL CommandLine:0 - System.InvalidOperationException: Unable to find symbol for node That inside function NHibernate.Test.EntityModeTest.Map.Basic.DynamicClassFixture.ShouldWorkWithLinqAndDynamics()
   à AsyncGenerator.Analyzation.Internal.ProjectAnalyzer.<ScanAllMethodReferenceLocations>d__78.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.Analyzation.Internal.ProjectAnalyzer.<ScanMethodData>d__70.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.Analyzation.Internal.ProjectAnalyzer.<ScanDocumentData>d__64.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.Analyzation.Internal.ProjectAnalyzer.<Analyze>d__30.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.AsyncCodeGenerator.<GenerateProject>d__2.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.AsyncCodeGenerator.<GenerateAsync>d__0.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à AsyncGenerator.CommandLine.Program.Main(String[] args)
```

**Comment by maca88** [comment]
I've added a basic support for `dynamic`, which supports the examples from your branch. I will release a new version after fixing #114.

**Comment by fredericDelaporte** [comment]
Great, thanks again maca88.

**Comment by maca88** [comment]
Released `0.13.0` and `0.8.2.8` which have the issues fixed. Do note that you have to add
```
- conversion: Ignore
  name: GetEnumerator
  containingTypeName: IFutureEnumerable
```
in the configuration due to [this change](https://github.com/maca88/AsyncGenerator/commit/eb5a2ffcab7a69d212a4b1cd5daf27223d77c240).
